src/parcsr_ls/par_mgr_interp.c

```diff
@@ -714,188 +718,182 @@ hypre_ExtendWtoPHost(HYPRE_Int      P_nr_of_rows,
 
          row_counter++;
       }
+
       /* update off-diagonal row pointer */
       P_offd_i[i + 1] = jj_counter_offd;
    }
    P_diag_i[P_nr_of_rows] = jj_counter;
 
    hypre_TFree(fine_to_coarse, HYPRE_MEMORY_HOST);
-   return 0;
+
+   return hypre_error_flag;
 }
 
 /*--------------------------------------------------------------------------
  * hypre_MGRBuildPHost
  *
  * Interpolation for MGR - Adapted from BoomerAMGBuildInterp
+ *
+ * This function computes:
+ *   - *Wp_ptr: Pointer to the constructed auxiliary matrix Wp
+ *              (NULL if method == 0, meaning Wp is an empty matrix)
+ *   - *P_ptr: Pointer to the constructed interpolation matrix P
+ *             (Always non-NULL if successful)
  *--------------------------------------------------------------------------*/
-
 HYPRE_Int
 hypre_MGRBuildPHost( hypre_ParCSRMatrix   *A,
```

Comment:
Is there a reason why we are changing the argument signature here? AFF and AFC are not always needed right?